### PR TITLE
implement NSString.boolValue

### PR DESCRIPTION
**Pull request by jpsim**
Gives same output as Standard Foundation for the same test values.

### CODE REVIEW

File: TestFoundation/TestNSString.swift [hunk 1]
```diff
@@ -34,6 +35,17 @@ class TestNSString : XCTestCase {
             ("test_FromMalformedNullTerminatedCStringInUTF8", test_FromMalformedNullTerminatedCStringInUTF8 ),
         ]
     }
+
+    func test_boolValue() {
+        let trueStrings: [NSString] = ["t", "true", "TRUE", "tRuE", "yes", "YES", "1"]
```
Comment (phausler) [inline TestFoundation/TestNSString.swift]: the value "+000009" should map to true

File: TestFoundation/TestNSString.swift [hunk 2]
```diff
@@ -34,6 +35,17 @@ class TestNSString : XCTestCase {
...
+        for string in trueStrings {
+            XCTAssert(string.boolValue)
+        }
+        let falseStrings: [NSString] = ["false", "FALSE", "fAlSe", "no", "NO", "0", "<true>", "_true"]
```
Comment (phausler) [inline TestFoundation/TestNSString.swift]: the value "-00000" should map to false

**Comment by harlanhaskins** [comment]
The NSString documentation says:

```
This property is YES on encountering one of "Y", "y", "T", "t", or a digit 1-9—the method ignores any trailing characters. This property is NO if the receiver doesn’t begin with a valid decimal text representation of a number.

The property assumes a decimal representation and skips whitespace at the beginning of the string. It also skips initial whitespace characters, or optional -/+ sign followed by zeroes.
```

It looks like, instead of checking the prefix for each character, it walks the string and checks if any of the characters match the predicates.
But it should also skip `+` or `-` and any number of 0's.

**Comment by jpsim** [comment]
Thanks for the tip @harlanhaskins! This one's on me, I should have referred to the documentation.

**Comment by jpsim** [comment]
I have an implementation in place, but I also had to implement `NSScanner.rangeOfCharacterFromSet (...)` which I'll package up in another PR.

**Comment by jpsim** [comment]
This has been updated to address @phausler's feedback and abide by the documentation referred to by @harlanhaskins. This depends on #75.

**Comment by harlanhaskins** [comment]
Looks great! :+1:

**Comment by jpsim** [comment]
Now that #75 has been merged, I've dropped that commit from this branch, so now this just has the `NSString.boolValue` commit.

**Comment by parkera** [comment]
Thanks!

**Pull request merged**